pkg/util/namespace.go

```diff
@@ -72,3 +88,12 @@ func EnsureNamespaceExist(client kubeclient.Interface, namespace string, dryRun
 
 	return createdObj, nil
 }
+
+// checkNamespaceAnnotations check whether the namespace is created by `karmadactl join`.
+// If not, we should not delete it.
+func checkNamespaceAnnotations(namespaceObj *corev1.Namespace) bool {
+	if annotationValue := namespaceObj.Annotations[namespaceAnnotationKey]; annotationValue == namespaceAnnotationValue {
```

Comment:
Need to check if `namespaceObj.Annotations` is nil.

Reply:
Done.